src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -605,6 +727,63 @@ public void withLock(boolean waitForRecovery, ChannelConsumer consumer) throws I
             }
         }
 
+        public void retryingPublishWithLock(ChannelConsumer consumer, ConnectionManagerConfiguration configuration) throws InterruptedException {
+            Iterator<RetryingDelay> iterator = configuration.createRetryingDelaySequence().iterator();
+            lock.lock();
+            try {
+                var currentValue = iterator.next();
```

Comment:
Shouldn't we do that inside the loop? From my understanding each retry should take the next value from that iterator